src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs

```diff
@@ -1804,5 +1804,231 @@ static void Main()
             CompileAndVerify(src1 + src2 + src3, expectedOutput: "C { Y = 22, X = 11, U = 44, Z = 33 }").VerifyDiagnostics();
             CompileAndVerify(new[] { src1, src2, src3 }, expectedOutput: "C { Y = 22, X = 11, U = 44, Z = 33 }").VerifyDiagnostics();
         }
+
+        [ConditionalFact(typeof(DesktopOnly))]
+        public void EqualityContractGetter_CompilerGeneratedAttribute()
+        {
+            var verifier = CompileAndVerify(@"
+using System;
+
+record C;
+");
+
+            verifier.VerifyTypeIL("C", @"
+.class private auto ansi beforefieldinit C
+    extends [mscorlib]System.Object
+    implements .custom instance void System.Runtime.CompilerServices.NullableAttribute::.ctor(uint8) = (
+        01 00 00 00 00
+    )
+    class [mscorlib]System.IEquatable`1<class C>
+{
+    .custom instance void System.Runtime.CompilerServices.NullableContextAttribute::.ctor(uint8) = (
+        01 00 01 00 00
+    )
+    .custom instance void System.Runtime.CompilerServices.NullableAttribute::.ctor(uint8) = (
+        01 00 00 00 00
+    )
+    // Methods
+    .method family hidebysig specialname newslot virtual 
+        instance class [mscorlib]System.Type get_EqualityContract () cil managed 
+    {
+        .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+            01 00 00 00
```

Comment:
While the attribute exists in the test, the following lines produce an empty `ImmutableArray<AttributeData>`:

```csharp
var member = verifier.Compilation.GetTypeByMetadataName("C").GetMembers("get_EqualityContract").Single();
var attributes = member.GetAttributes();
```

I have no idea why.

Reply:
Found that other tests operate on metadata symbols when working with `GetAttributes`. I updated the test, though I'm interested in understanding more on that if someone have the time :)

Reply:
> though I'm interested in understanding more on that if someone have the time

For symbols from source, ```GetAttributes``` never includes attributes that compiler is going to add itself, its result reflects what is in source. The ```symbolValidator``` operates on symbols that we load from the emitted binary for validation purposes. ```GetAttributes``` for those symbols reflects all attributes present in metadata, i.e. all attributes that were actually emitted.